libs/newsletters-data-client/src/lib/zod-helpers/user-data-schema.ts

```diff
@@ -0,0 +1,60 @@
+import type {ZodObject, ZodRawShape} from "zod";
+import {newsletterDataSchema} from "../schemas/newsletter-data-type";
+import type { UserPermissions } from '../user-profile';
+
+export const getUserEditSchema = (permissions: UserPermissions): ZodObject<ZodRawShape> => {
+	const { editBraze, editOphan, editTags, editSignUpPage, editNewsletters } =
+		permissions;
+	if (editNewsletters) {
+		return newsletterDataSchema.pick({
+			name: true,
+			frequency: true,
+			regionFocus: true,
+			theme: true,
+			status: true,
+			tagCreationsStatus: true,
+			seriesTag: true,
+			composerTag: true,
+			composerCampaignTag: true,
+			ophanCampaignCreationsStatus: true,
+			signupPageCreationsStatus: true,
+			signupPage: true,
+			signUpDescription: true,
+			brazeCampaignCreationsStatus: true,
+			brazeNewsletterName: true,
+			brazeSubscribeAttributeName: true,
+			brazeSubscribeEventNamePrefix: true,
+			brazeSubscribeAttributeNameAlternate: true,
+		});
+	}
+	if (editBraze) {
+		return newsletterDataSchema.pick({
+			brazeCampaignCreationsStatus: true,
+			brazeNewsletterName: true,
+			brazeSubscribeAttributeName: true,
+			brazeSubscribeEventNamePrefix: true,
+			brazeSubscribeAttributeNameAlternate: true,
+		});
+	}
+	if (editOphan) {
+		return newsletterDataSchema.pick({
+			ophanCampaignCreationsStatus: true,
+		});
+	}
+	if (editTags) {
+		return newsletterDataSchema.pick({
+			tagCreationsStatus: true,
+			seriesTag: true,
+			composerTag: true,
+			composerCampaignTag: true,
+		});
+	}
+	if (editSignUpPage) {
+		return newsletterDataSchema.pick({
+			signupPageCreationsStatus: true,
+			signupPage: true,
+			signUpDescription: true,
+		});
+	}
+	throw new Error('No permissions found');
```

Comment:
Do we want this to throw an Error when the user doesn't have any permissions? Might that cause a run time error when it is used in a component? 

Would it work to return an empty schema as a fallback - IE  `newsletterDataSchema.pick({})` - that way the function should be safe and would reflect that there are no fields the user is allowed to edit

Reply:
Yep - much better